src/main/scala/com/typesafe/sbt/SbtScalariform.scala

```diff
@@ -31,69 +31,48 @@ object SbtScalariform
   override def requires = plugins.JvmPlugin
   override def trigger = allRequirements
 
-  override def projectSettings = {
-    // workaround autoplugin defaults overriding user settings
-    // see https://github.com/sbt/sbt/issues/3389
-    if (!processed.get) {
-      autoImport.scalariformSettings(autoformat = true)
-    }
-    else Seq.empty
-  }
-
-  private val processed =
-    new java.util.concurrent.atomic.AtomicBoolean(false)
-
   object autoImport {
     val scalariformFormat = taskKey[Seq[File]]("Format (Scala) sources using scalariform")
     val scalariformPreferences = settingKey[IFormattingPreferences]("Scalariform formatting preferences")
+    val scalariformAutoformat = settingKey[Boolean]("Whether Scala sources should be auto formatted when compile is run")
+    val scalariformDoAutoformat = taskKey[Seq[File]]("Format sources if autoformat is configured")
+
+    @deprecated("Use scalariformAutoformat to turn autoformat on or off", "1.8.1")
+    def scalariformSettings(autoformat: Boolean): Seq[Setting[_]] = Nil
 
-    def scalariformSettings(autoformat: Boolean): Seq[Setting[_]] = {
-      defaultSettings(autoformat, withIt = false)
-    }
+    @deprecated("Use scalariformItSettings", "1.8.1")
+    def scalariformSettingsWithIt(autoformat: Boolean): Seq[Setting[_]] = scalariformItSettings
 
-    def scalariformSettingsWithIt(autoformat: Boolean): Seq[Setting[_]] = {
-      defaultSettings(autoformat, withIt = true)
-    }
+    def scalariformItSettings: Seq[Setting[_]] = inputs(It) ++ inConfig(It)(configScalariformSettings)
   }
   import autoImport._
 
+  override def globalSettings = Seq(
+    scalariformPreferences := defaultPreferences,
+    includeFilter in scalariformFormat := "*.scala",
+    scalariformAutoformat := PreferencesFile(None).forall(_.autoformat.autoformat)
+  )
```

Comment:
Not sure that Global is the right scope for these, they probably should be defined at the build scope.

Reply:
> Not sure that Global is the right scope for these

Let's leave as is for now, can look into changing before publishing a release -- have to sort out another issue with Play generated sources (router *.scala files) being reformatted on every `compile`, and then we should be good for 1.8.2

Reply:
Really? As far as I can see it's only including `unmanagedSourceDirectories`, which should mean the router files aren't reformatted, unless the Play routes compiler is adding its output directory to `unmanagedSourceDirectories`, which would be really bad if it is.

Reply:
It was [another change](https://github.com/sbt/sbt-scalariform/commit/42ca0f04b5e7280afcf8eec896580f141be85891), spurred on by users wanting to have sources in project root (including build def) formatted.

Unfortunately generated play router sources also live in the root (compile target), which causes scalariform to format the sources, which in turn causes Play to recompile the sources as they've changed since last compilation.

Apparently one possible fix is using `ThisProject` in place of `LocalRootProject`.

Reply:
The use of global scope is perfectly fine for plugins to use as long as it's on your own key or task scoping, and there is zero chance of conflicting with either the default settings provided by others (like sbt's JvmPlugin).

Reply:
Is global scope basically `ThisBuild` + whatever `~/.sbt/<version>/*.sbt` defines? Or can global scope also impact settings in other build units? If it's the former, then global scope is probably the right idea - it means that users can add scalariform to their global plugins and configure it there (though there's probably many down sides of doing that).

Reply:
As far as I know, global scope just means `in Global` aka `in (Zero, Zero, Zero)` for all axes, so just strictly weakest scope. In practice, I don't think it matters all that much if a plugin used `buildSettings` or `globalSettings`, because to make a difference we'd need to use source-dependency where `{some-uri}/scalariformPreferences ` value is set differently from `{.}/scalariformPreferences `.